src/Components/Aspire.Azure.Messaging.EventHubs/EventProcessorClientComponent.cs

```diff
@@ -0,0 +1,93 @@
+            (options, cred) =>
+            {
+                var connectionString = settings.ConnectionString;
+                if (string.IsNullOrEmpty(connectionString) && string.IsNullOrEmpty(settings.Namespace))
+                {
+                    throw new InvalidOperationException(
+                        $"A EventHubProducerClient could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'ConnectionString' or 'Namespace' in the '{configurationSectionName}' configuration section.");
+                }
+
+                if (string.IsNullOrEmpty(settings.EventHubName))
+                {
+                    throw new InvalidOperationException(
+                        $"A EventProcessorClient could not be configured. Ensure a valid EventHubName was provided in the '{configurationSectionName}' configuration section.");
+                }
+
+                options.Identifier ??= GenerateClientIdentifier(settings);
+
+                var blobClient = GetBlobContainerClient(settings, cred, configurationSectionName);
+
+                var processor = !string.IsNullOrEmpty(connectionString)
+                    ? new EventProcessorClient(blobClient,
+                        settings.ConsumerGroup ?? EventHubConsumerClient.DefaultConsumerGroupName, connectionString)
+                    : new EventProcessorClient(blobClient,
+                        settings.ConsumerGroup ?? EventHubConsumerClient.DefaultConsumerGroupName, settings.Namespace,
+                        settings.EventHubName, cred, options);
+
+                return processor;
+
+            }, requiresCredential: false);
+    }
+
+    private BlobContainerClient GetBlobContainerClient(
+        AzureMessagingEventHubsSettings settings, TokenCredential cred, string configurationSectionName)
+    {
+        if (string.IsNullOrEmpty(settings.BlobClientConnectionName))
+        {
+            // throw an invalid operation exception if the blob client connection name is not provided
+            throw new InvalidOperationException(
+                $"A EventProcessorClient could not be configured. Ensure a valid connection name was provided in the '{configurationSectionName}' configuration section.");
+        }
+
+        var blobConnectionString =
+            builderConfiguration.GetConnectionString(
+                settings.BlobClientConnectionName) ??
+                throw new InvalidOperationException(
+                    "A EventProcessorClient could not be configured. " +
+                    $"There is no connection string in Configuration with the name {settings.BlobClientConnectionName}. " +
+                    "Are you sure you defined a Blob resource with this name in your AppHost?");
+
+        // FIXME: ideally this should be pulled from services; but thar be dragons.
+        // There is no reliable way to get the blob service client from the services collection
```

Comment:
Why not? Why not grab the `BlobContainerClient` or `BlobServiceClient` from DI, and if it isn't there, throw? That's the whole point of DI... to inject dependencies.

Reply:
There is no service provider in scope here. Also, if it was added without a ServiceKey, how do I know if it's the one I need? This needs more thought.

Reply:
I am going to remove the blob component code from the worker in this sample as it is actually doing nothing for these very reasons except seed confusion. I agree @eerhardt that we _should_ be using the DI instance if it's there, but the plumbing isn't there yet.

Reply:
Also, pulling from DI (assuming a service provider is available) puts an ordering dependency on the components that may not be obvious without clear documentation.

Reply:
removed blob component from sample eventhub (but it remains as a resource in apphost)

Reply:
Can you open an issue for this? I think this is going to need to be reworked before this component can be GA'd. Developers are going to want to configure this Blob client more than just passing the "connection name" (which according to the current code can only be a URI).

These components need to be "composable". That's the whole point.